perun-core/src/main/java/cz/metacentrum/perun/core/entry/UsersManagerEntry.java

```diff
@@ -508,6 +509,26 @@ public UserExtSource updateUserExtSource(PerunSession sess, UserExtSource userEx
 		return getUsersManagerBl().getUserExtSources(sess, user);
 	}
 
+	@Override
+	public List<RichUserExtSource> getRichUserExtSources(PerunSession sess, User user) throws InternalErrorException, UserNotExistsException, PrivilegeException {
+		Utils.checkPerunSession(sess);
+
+		// Authorization
+		if(!AuthzResolver.isAuthorized(sess, Role.VOADMIN) &&
+			!AuthzResolver.isAuthorized(sess, Role.VOOBSERVER) &&
+			!AuthzResolver.isAuthorized(sess, Role.GROUPADMIN) &&
+			!AuthzResolver.isAuthorized(sess, Role.ENGINE) &&
+			!AuthzResolver.isAuthorized(sess, Role.RPC) &&
+			!AuthzResolver.isAuthorized(sess, Role.SELF, user) &&
+			!AuthzResolver.isAuthorized(sess, Role.PERUNOBSERVER)) {
+			throw new PrivilegeException(sess, "getRichUserExtSources");
+		}
+
+		getUsersManagerBl().checkUserExists(sess, user);
+
+		return getUsersManagerBl().getRichUserExtSources(sess, user);
```

Comment:
We should implement ``filterOnlyAllowedAttributes()`` method to perform authorization check on list of ``RichUserExtSource`` as we have for groups and members. It will be part of Bl layers and called from here.

Also, I don't know if we have authorization check on ues attributes. I believe, that right now they are accessible only to perun admin, and will need to configure it on each instance on deployment.